scripts/setup/postgres-init-db

```diff
@@ -2,32 +2,74 @@
 set -e
 set -x
 
+# A bit of a helper variable, default is voyager
+DEPLOYMENT_TYPE="${DEPLOYMENT_TYPE:-voyager}"
+# What user should we use for connecting to the database
+POSTGRES_USER="${POSTGRES_USER:-postgres}"
+
 # Shut down all services to ensure a quiescent state.
-supervisorctl stop all
+# Stop supervisord only if the socket exists, prevents problems with docker
+if [ -e "/var/run/supervisor.sock" ]; then
+    supervisorctl stop all
+fi
+
+# Don't "terminate" psql sessions when run in docker!
+if [ "$DEPLOYMENT_TYPE" != "dockervoyager" ]; then
+    # Drop any open connections to any old database.  Hackishly call using
+    # source because postgres user can't read /root/zulip/scripts/setup.
+    source "$(dirname "$0")/terminate-psql-sessions" postgres zulip zulip_base
+fi
```

Comment:
OK on the supervisor front I think it's reasonable to check for the socket -- that's a correct check for whether supervisord is running.

For the `terminate_psql_sessions` part, I'm not sure why that would be a problem in the docker setting -- the next lines in this script only work if the script can't access to the database anyway...